cdap-formats/src/test/java/io/cdap/cdap/format/StructuredRecordBuilderTest.java

```diff
@@ -285,4 +285,25 @@ public void testInvalidPrecision() {
     Schema schema = Schema.recordOf("test", Schema.Field.of("d", Schema.decimalOf(5, 2)));
     StructuredRecord.builder(schema).setDecimal("d", new BigDecimal(new BigInteger("12341324"), 2)).build();
   }
+
+  @Test(expected = ClassCastException.class)
```

Comment:
It would be great to test the fix: not only exception type, but message as well. You should be able to use https://junit.org/junit4/javadoc/latest/org/junit/rules/ExpectedException.html to do it.

Reply:
nice, did not know about this, will update